packages/@aws-cdk/cdk/lib/core/construct.ts

```diff
@@ -532,6 +533,15 @@ export class ConstructNode {
     return ret;
   }
 
+  /**
+   * Return the path of components up to but excluding the root
+   */
+  public rootPath(): IConstruct[] {
```

Comment:
Maybe we can just delete this. Is this really a useful thing? I also don't like that it's called `path` because it doesn't return a path (paths are strings), it returns an array of ancestors without the first one, which brings me to ask, what's rong with `construct.node.ancestors().slice(1)`

Reply:
Can you also make ancestors a property please?

Reply:
"root path" is the term for the nodes in a tree that lie between a given node and the root of the trie. I wouldn't say a path *is* a string, maybe that a path can be represented as a string.

`rootPath` could (and should) have more logic for the purposes of IDs, but honestly our tree structure and which things exist and have ids and whether those ids should be taken into account is completely messed up and inconsistent between real applications and unit tests, insofar as that anything that deals with names and paths is a tangle of messy ifs.

So sure, I'll inline the definition of `rootPath()` everywhere.

`ancestors()` takes an optional argument, so it cannot be made a property.